### ISSUE TITLE
Is Silence still alive?

### ISSUE THREAD

**Issue by whiver** [SilenceIM/Silence#703]
This is a simple question, but seeing the last commit going back to March, the amount of issues increasing over time and the few pull requests open not merged for a while, I am asking if Silence is still maintained.

I understand that it's not easy to maintain such a project, especially on the contributors' free time.
Though, seeing the PR not merged since March does not encourage contributions either.
The app has still several bugs that are quite frustrating and it would be really nice to have them fixed one day, so that's why I'm asking you what we can expect from the development, and if it is still time to contribute!

Thanks :slightly_smiling_face:

**Comment by 0pLuS0** [comment]
I'd like to know too?

I've always thought/assumed, when it comes to security, there is a lot of constant development going on, and seeing nothing for almost 4 months now, has me concerned.

Hmm

**Comment by johanw666** [comment]
I see no direct threat to the security of Silence, the only thing that can currently break the encryption in transit is a working quantum computer with many qbits but if such a machine is built a lot of public key crypto should be adapted.

When the device if physically compromised there are of course a lot of things that could happen around the encryption. Signal is trying to address these issues with their move to SQLCipher (breaking backup tools in the process) but I think no perfect solution for that problem exists.

**Comment by whiver** [comment]
The FAQ draft from the issue #686 says that:
> Apple doesn't allow third apps like Silence to manage SMS/MMS messages. This means that you cannot use other apps to chat on iOS than the stock SMS/MMS one. Messaging apps on iOS rely on Internet data rather than on GSM.

So the answer is no 🙂

**Comment by 0pLuS0** [comment]
The iOS question was off topic... 😛

Well I think I'll just have to get everyone on Android...

**Comment by rugk** [comment]
Any news here?

**Comment by 0pLuS0** [comment]
I don't want to go off topic here, but it would be great if the DEV(s) or someone that seriously knows could chime in on this conversation in regards to using an app like this.

What I mean, is I've come to realize, of course depending on where you live, that some cellphone providers don't even store any data on consumers, but there are some that do. So I can see Silence as an advantage when using a provider that stores all the text and pic messages you send, otherwise, what are we concerned here with MITM attacks, someone snooping/sniffing in on our conversations? Because that is really going to take some effort and work on someone's part to do, and you really better be worth all the trouble.

So bottom line, get with a private cellphone provider that protects your data and doesn't store anything, and then, I'm not getting the point of using Silence, becacuse I was more concerned with companies storing data and spying on us, over someone trying a MITM attack...

Hmm

**Comment by rugk** [comment]
> So bottom line, get with a private cellphone provider that protects your data and doesn't store anything

If we could or we were sure they are not, we very likely would not use Silence.

> I'm not getting the point of using Silence

Then why do you comment here? If you do not want to use silence then don't. There is no problem in that.

>  I was more concerned with companies storing data and spying on us, over someone trying a MITM attack

So that is your actual point? Protecting against MITM attacks is pointless?

If so, then yes, to some extend I can agree to you. Of course, providers may not really do MITM attacks, usally. And it is much more realistic that they just store the SMS data. This is both easier and harder to detect.
So this is all about thread models. So first: It's perfectly fine, if you have a different thread model. Choose, what suits you best.
However, in case of Silent, I think you need to extend your thread model, because Silence protection (e2e encryption) helps in many cases. Here are some:
1. You may choose a "good" provider yes, but SMS involves two parties and two providers (at least, don't know whether stuff may be routed around even more). As such, the person your are texting also has another provider.
2. You may have heard of [about IMSO catchers](https://en.wikipedia.org/wiki/IMSI-catcher). This is a kind of MITM attack, altghough, of course, not on your phone provider, but on the connection between the cellphone tower and you. Also, it very likely that they won't try to impersonate/MITM the Silence e2e encryption, but without Silence they can just sniff the connection.

Apart from that every good e2e encryption protocol should protect against MITM attacks. Just because it would not be sufficient/complete otherwise and be thus attackable. In case of Silence that may not even be done in the best way, due to the not existing indicator for verified contacts. (see https://github.com/SilenceIM/Silence/issues/210)

----

However, generally...

> I don't want to go off topic here

But you really got.
As such continue in https://github.com/SilenceIM/Silence/issues/210. This issue here is about a general reply on the development progress.

**Comment by 0pLuS0** [comment]
I wasn't attacking being negative about using Silence, just bring up some points is all...

But you make one very valid point, the person on the other end receiving and sending texts, but then if they don't have, don't want to be bothered with Silence, like trying to convince the Noobs to use it, well we're screwed then too.

Anyhow, I'm done. :)

**Comment by ArchangeGabriel** [comment]
I’ve been able to contact @BLeQuerrec. He is very busy (and has no time to come here on GH), but still works on Silence when he gets some free time. His first priority currently is to upgrade the used Android API in order to meet the [November 2018 API level 26 mandatory target deadline](https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html).

So Silence is still alive. ;)

**Comment by 1anomaly1** [comment]
I'm very gratefull for Silence. It is so important that Android has an app like this to advance the causes of Freedom, without Google, etc. This app is at the forefront of this war. i encourage the developer, and everyone else with legitimate interested, to put these kinds of FOSS, and Linux over their friends, and to view anything apple, especially friends, as a lost cause.

**Comment by boggle247** [comment]
@ArchangeGabriel well the last change as of today is still March of 2018, so it appears that he missed that deadline? What are the ramifications for that?

**Comment by licaon-kter** [comment]
@boggle247 nope, F-Droid does not impose anything, but then again, the build tools still come from Google and those have timebombs in them, that you need to defuse...

**Comment by ArchangeGabriel** [comment]
For now it just means that simple updates would not be allowed to go through (the PlayStore, F-Droid has no such requirements). As time passes, upgrades to be done for being allowed to update in PlayStore will be more and more important, and thus require more work. Same goes for backporting upstream (Signal) changes.

I still believe @BLeQuerrec wants to continue developing Silence, but lacks time to do so. It would be nice to find more people with knowledge of Java & Android interested in developing this project further, but the interest is low because nowadays most people use data to communicate, so that solutions like Briar are way superior and much more attractive. I still use Silence because I have lots of contacts who don’t have data at all, but I guess this is going to be more and more the exception.

**Comment by whiver** [comment]
IMO Silence has a real advantage over solutions like Briar or Riot, being that it uses plain SMS to communicate, allowing users to send message to everyone with the same app: encrypted for those who also use Silence, unencrypted for the others, whereas with a solution like Briar you'll have to convince everyone to use your app if you want to communicate with them, which is often hard to do.
But that's not the subject: if we are here to discuss the future of Silence, that means that there are still people interested in it :slightly_smiling_face: 

I think we mainly need someone reviewing and merging the PRs, that would encourage contributors.

**Comment by tuxayo** [comment]
On concrete way to show interest is to donate in the recurring donation campaign of Silence: https://liberapay.com/Silence/

@BLeQuerrec needs to setup a Stripe[1] or Paypal account since the big Liberapay changes last year.

Pledging on the page is still useful. We will be notified we the payment will be possible.

[1] less fees, at least for me from France

**Comment by maxfunky** [comment]
The warrant canary was last signed 1 year ago "https://silence.im/canary.txt"

**Comment by ArchangeGabriel** [comment]
Well I wouldn’t worry too much about the canary for now since no version has been released since…

**Comment by thiswillbeyourgithub** [comment]
@whiver also, silence can be used in remote places. I can't think of a better way to ue encryption in remote africa for journalists for example (on the top of my head)

**Comment by rugk** [comment]
Anyway, it would be good to see some development again. The last thing we've heard that they have some big ideas and even wanted to integrate XMPP and so on…
IMHO, these were probably too big and I would be totally fine with a proper, but maintained SMS-encryption-based messaging application than needing big fancy new features…

**Comment by keikoro** [comment]
I switched back to Signal in 2018 and had everyone else I had used Silence with switch as well when the missing MMS export feature became too big an obstacle; I'd still like to see this if nothing else fixed, so a full migration without data loss is possible (I have kept Silence around for just this reason but really want to finally get rid of it).

**Comment by Chewie9999** [comment]
Is there a way to donate without having to use bitcoin (i don't have any) or Librepay (which still is saying a payment methods is not available)?
Yes, it's a niche app, but i still feel it is important to have it, and I use it daily.

**Comment by ArchangeGabriel** [comment]
As I stated [here](https://github.com/SilenceIM/Silence/issues/747#issuecomment-493526800), the project is still alive and some news should be expected not too long from now.

@Chewie9999 I don’t understand your issue with Liberapay, it should work (they accept SEPA direct debit if you live in € zone, or anything that Stripe or PayPal accepts otherwise).

**Comment by strypey** [comment]
In #747 @ArchangeGabriel wrote:
> some progresses have been made and should be released in the coming weeks or so.

Is it possible to have a look at the code changes due to be released? Are they being worked on in a fork or something?

**Comment by ArchangeGabriel** [comment]
@strypey Not currently, but it will of course be available before release. The main reason being development moving away from GitHub to a self-hosted GitLab instance, but which is not publicly available for now.

**Comment by Chewie9999** [comment]
> As I stated [here](https://github.com/SilenceIM/Silence/issues/747#issuecomment-493526800), the project is still alive and some news should be expected not too long from now.
> 
> @Chewie9999 I don’t understand your issue with Liberapay, it should work (they accept SEPA direct debit if you live in € zone, or anything that Stripe or PayPal accepts otherwise).

On the liberapay site, it says the following:
" Silence hasn't configured any payment method yet, so your donation cannot actually be processed right now. We will notify you when payment becomes possible."
Therefore I can send money, but at the moment, the devs can't get the money out, so my donation won't get to them.

**Comment by ArchangeGabriel** [comment]
OK indeed, I’ve checked with the dev and that is because it does not want to have neither a Stripe nor a PayPal account, which I definitively understand. He will be looking for other options once the next release will be out.

**Comment by tuxayo** [comment]
> does not want to have neither a Stripe nor a PayPal account

Could a trusted collaborator who already has such an account could forward the payments? I'm afraid that even when offering other politically better funding options, not having one of those two would exclude many people due to currency or inability to access the other funding options. Which could still be advertised as preferred to encourage transition to better payment providers.

**Comment by strypey** [comment]
@tuxayo seems like you need to talk to the Liberapay folks about offering devs more options for taking payment than just PayPal and Stripe. This is more complicated than you'd think though. Many countries now have all sorts of financial regulations that purport to be about preventing wire fraud, tax evasion, money laundering, drug running, financing of terrorism etc. [GNU Taler](https://taler.net/) is an attempt to build an online payment system that is free code software, and both respects customer privacy *and* allows regulators to hold vendors accountable, but it can only work if they can find banks willing to work with Taler instances.

**Comment by strypey** [comment]
@ArchangeGabriel 
> The main reason being development moving away from GitHub to a self-hosted GitLab instance, but which is not publicly available for now.

Any updates on this?

**Comment by BLeQuerrec** [comment]
https://github.com/SilenceIM/Silence/issues/753#issuecomment-516117577

**Comment by thiswillbeyourgithubiscia** [comment]
I agree.
